### ISSUE TITLE
REST API error 401

### ISSUE THREAD

**Issue by dldnet** [YetiForceCompany/YetiForceCRM#12436]
Hi,
I try to connect the REST API of YetiForce 5.2 self hosted
I use yetirest.php from GitHub.
I have defined the server and user properties
![image](https://user-images.githubusercontent.com/42601108/72662250-33f1cc80-39e5-11ea-9ee3-46d9f3b0a1a9.png)
![image](https://user-images.githubusercontent.com/42601108/72662260-4d931400-39e5-11ea-8d61-f6182be8d3f7.png)

And here is the result of the script:
![image](https://user-images.githubusercontent.com/42601108/72662275-96e36380-39e5-11ea-9702-f0f3a1caa897.png)

I have searched in the issues and tried many options but nothing... always a 401 error

Do you have anything to try ?

Thanks in advance

**Comment by mariuszkrzaczkowski** [comment]
![github](https://user-images.githubusercontent.com/10198654/72667689-ce223680-3a1e-11ea-8a73-98426a0d90a1.jpg)

**Comment by mariuszkrzaczkowski** [comment]
Logs

**Comment by dldnet** [comment]
[webserviceErrors.log](https://github.com/YetiForceCompany/YetiForceCRM/files/4085417/webserviceErrors.log)
[system.log](https://github.com/YetiForceCompany/YetiForceCRM/files/4085419/system.log)
[webserviceDebug.log](https://github.com/YetiForceCompany/YetiForceCRM/files/4085420/webserviceDebug.log)

I don't know if it's a bug or a problem with my config. 
I didn't found any trace in the logs except 401 error. 
I am sure of my passwords (really simple passwords for testing)

**Comment by dldnet** [comment]
OS: CentOs 6.10
PHP 7.3 for both (YetiForce and therequesting script)
Both on the same server to test

**Comment by mariuszkrzaczkowski** [comment]
screenshot of configuration verification

**Comment by dldnet** [comment]
![image](https://user-images.githubusercontent.com/42601108/72715988-c8893580-3b71-11ea-9cc4-e19e51b23b0e.png)

**Comment by dldnet** [comment]
![yetiforce_settings](https://user-images.githubusercontent.com/42601108/72716232-2ae23600-3b72-11ea-8576-a191044d4c66.png)

**Comment by mariuszkrzaczkowski** [comment]
> [webserviceErrors.log]
> [system.log]
> [webserviceDebug.log]
> 
> I don't know if it's a bug or a problem with my config.
> I didn't found any trace in the logs except 401 error.
> I am sure of my passwords (really simple passwords for testing)

send full logs to github@yetiforce.com, because they contain sensitive data

**Comment by dldnet** [comment]
mail sent.
Thanks

**Comment by rskrzypczak** [comment]
I think that authorization header is not passed to PHP script
Check: https://github.com/YetiForceCompany/YetiForceCRM/blob/developer/public_html/.htaccess#L21

**Comment by dldnet** [comment]
Thanks for the info. I have tried but nothing changed.
Still 401 error, authorization problem

**Comment by dldnet** [comment]
Any idea ?

**Comment by mariuszkrzaczkowski** [comment]
The data in the authorization header is incorrect, enter the data from Web service - Applications instead of Web service - users

**Comment by dldnet** [comment]
That's exactly what I did since the beginning

**Comment by dldnet** [comment]
Ok, first problem found.
Apache was dropping headers (including Authentication).
It was due to server config.
Now I have setup PHP-FPM on the YetiForce server and headers are well sent to php.
No more 401 error...
But now: Error 500 - internal server error
And it is far more difficult to debug

**Comment by mariuszkrzaczkowski** [comment]
enable all logs and you'll see an error message

**Comment by rskrzypczak** [comment]
Enable https://github.com/YetiForceCompany/YetiForceCRM/blob/012eda20637db05d548f1ca173e21983e83a3ef0/config/Debug.php#L88

**Comment by dldnet** [comment]
I have already activated debug. When it is activated, no more error 500 (seems logical because the server answers something). But I receive a REST error empty/
This is what I see in the logs:

Status: 200
REQUEST_METHOD: POST
REQUEST_URI: /webservice/Users/Login
QUERY_STRING: module=Users&action=Login
PATH_INFO: 
----------- Response data -----------
Array
(
    [status] => 0
    [error] => Array
        (
            [message] => 2: openssl_public_encrypt() expects parameter 1 to be string, array given in /****/******/YetiForceCRM/api/webservice/Core/Response.php, line 132
            [code] => 200
        )

)



So, I tried to call the REST service via http (in place of https) but nothing changed.
